Document modals & input text (#4143)
hammerandchisel/discord-api-docs: docs/interactions/Message_Components.md

| min_length?  | int                                                                                                         | title of modal                                                                            | [Input Text](#DOCS_INTERACTIONS_MESSAGE_COMPONENTS/input-text)                                                                                                                                                                                                        |
| max_length?  | int                                                                                                         | title of modal                                                                            | [Input Text](#DOCS_INTERACTIONS_MESSAGE_COMPONENTS/input-text)                                                                                                                                                                                                        |

> | placeholder? | string                                                                                                      | title of modal                                                                            | [Modal](#DOCS_INTERACTIONS_MESSAGE_COMPONENTS/input-text)                                                                                                                                                                                                             |

Author: rip

> | placeholder? | string                                                                                                      | the placeholder for the text input field                                                  | [Input Text](#DOCS_INTERACTIONS_MESSAGE_COMPONENTS/input-text)                                                                                                                                                                                                        |

Reviewer: can't it be in the `placeholder?` above? or it doesn't have the limit of 100 characters?

Author: Ah you're right. Fixing this!

###### Component Types

| Type | Name        | Description                                |
| ---- | ----------- | ------------------------------------------ |
| 1    | Action Row  | A container for other components           |
| 2    | Button      | A button object                            |
| 3    | Select Menu | A select menu for picking from choices     |

> | 4    | Input Text  | A text box for inserting written responses |

Reviewer 2: ```suggestion
| 4    | Input Text  | A text box for inserting written responses in modals |
```
Otherwise it looks to me like you can also use these for regular messages

Author: Ah yes, thanks!

Input text is a modal component accepting text input from users.

- `input_text` must be sent inside an Action Row
- An Action Row can have exactly 1 `input_text` component

### Input Text Object

###### Input Text Structure

| Field        | Type                                                | Description                                                                                           |
| ------------ | ------- | ----------------------------------------------------------------------------------------------------- |
| type         | integer | `4` for a text input field                                                                            |
| style        | integer | one of [input text styles](#DOCS_INTERACTIONS_MESSAGE_COMPONENTS/input-text-object-input-text-styles) |
| label        | string  | text that appears on top of the input text field, max 80 characters                                   |
| placeholder? | string  | placeholder for the text input                                                                        |
| min_length?  | integer | minimal length of text input                                                                          |

> | max_length?  | integer | maximal length of text input                                                                          |

Reviewer 3: Why custom_id isn't here, but it's in the json sample?

Author: Thank you, you're right.
Fixed that + the description of `custom_id` on the select component fields.

When a user clicks on a button, your app will receive an [interaction](#DOCS_INTERACTIONS_RECEIVING_AND_RESPONDING/interaction-object) including the message the button was on:

## Input Text

Input text is a modal component accepting text input from users.

- `input_text` must be sent inside an Action Row
- An Action Row can have exactly 1 `input_text` component

### Input Text Object

###### Input Text Structure

> | Field        | Type                                                | Description                                                                                           |

Reviewer 3: ```suggestion
| Field        | Type    | Description                                                                                           |
```
It isn't aligned correctly.

When a user clicks on a button, your app will receive an [interaction](#DOCS_INTERACTIONS_RECEIVING_AND_RESPONDING/interaction-object) including the message the button was on:

## Input Text

Input text is a modal component accepting text input from users.

- `input_text` must be sent inside an Action Row

> - An Action Row can exactly 1 `input_text` components

Reviewer 4: ```suggestion
- An Action Row can have exactly 1 `input_text` component
```

| type         | integer | `4` for a text input field                                                                            |
| style        | integer | one of [input text styles](#DOCS_INTERACTIONS_MESSAGE_COMPONENTS/input-text-object-input-text-styles) |
| label        | string  | text that appears on top of the input text field, max 80 characters                                   |
| placeholder? | string  | a developer-defined identifier for the button, max 100 characters                                     |
| min_length?  | int     | minimal text input                                                                                    |
| max_length?  | int     | maximal length of text input                                                                          |

Input text components come in two styles for short or longer texts.

###### Input Text Styles

| Name      | Value | Description                         |
| --------- | ----- | ----------------------------------- |
| Short     | 1     | Intended for short single-line text |
| Paragraph | 2     | Intended for much longer inputs     |

###### Sample Input Text Component
```json
{

>     type: 4.

Reviewer 5: ```suggestion
    type: 4,
```

Typo?

Reviewer 5: And why this is not json

When a user clicks on a button, your app will receive an [interaction](#DOCS_INTERACTIONS_RECEIVING_AND_RESPONDING/interaction-object) including the message the button was on:

## Input Text

Input text is a modal component accepting text input from users.

> - `input_text` must be sent inside an Action Row

Reviewer 6: ```suggestion
- `input_text` must be sent inside an Action Row.
```

> - An Action Row can have exactly 1 `input_text` component

Reviewer 6: ```suggestion
- An Action Row can have exactly 1 `input_text` component.
```
